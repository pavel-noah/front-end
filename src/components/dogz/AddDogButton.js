import React from 'react';
import PropTypes from 'prop-types';

export default function AddDogButton({ submitForm, onChange, name, age, weight }) {

    return (<>
            <form onSubmit={event => submitForm(event, name, age, weight)}>
                Name: <input onChange={onChange} type="text" name="name" value={name}></input>
                Age: <input onChange={onChange} type="text" name="age" value={age}></input>
                Weight: <input onChange={onChange} type="text" name="weight" value={weight}></input>
                <button>submit dog</button>
            </form>
        </>
    );
}

AddDogButton.propTypes = {
    submitForm: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    name: PropTypes.string.isRequired,
    age: PropTypes.string.isRequired,
    weight: PropTypes.string.isRequired
};
