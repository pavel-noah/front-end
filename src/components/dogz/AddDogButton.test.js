import React from 'react';
import { shallow } from 'enzyme';
import AddDogButton from './AddDogButton';

describe('Add dog button component', () => {
    it('renders add dog button', () => {
        const wrapper = shallow(<AddDogButton submitForm={() => {}} onChange={() => {}} name="" age="" weight="" />);
        expect(wrapper).toMatchSnapshot();
    });
});
  
