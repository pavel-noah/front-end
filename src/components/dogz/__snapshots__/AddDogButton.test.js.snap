// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add dog button component renders add dog button 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    Name: 
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    Age: 
    <input
      name="age"
      onChange={[Function]}
      type="text"
      value=""
    />
    Weight: 
    <input
      name="weight"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button>
      submit dog
    </button>
  </form>
</Fragment>
`;
